src/project-css/app/Helpers/CMSHelper.php

```diff
@@ -0,0 +1,81 @@
+<?php
+/**
+ * @author Ajay Krishna Teja Kavur
+ * @author Tracy A McCormick <tam0013@mail.wvu.edu>
+ */
+
+namespace App\Helpers;
+
+use App\Models\CMSRecords;
+use App\Models\Collection;
+use App\Helpers\CSVHelper;
+use App\Helpers\TableHelper;
+
+class CMSHelper {
+    /**
+     * CMS Helper
+     *
+     * These are various functions that help with processing
+     * cms files.
+     *
+     */
+
+
+     /**
+      * Check the Record Types table for a matching header for the table
+      *
+      * @param integer $collctnID Contains the collection ID
+      * @param string $recordType Contains the Record type 1A, 1B, etc.
+      * @param integer $fieldCount Contains number of fields expected for the Record
+      *
+      * @author Tracy A. McCormick <tam0013@mail.wvu.edu>
+      * @return unserialized Array if found or null
+      */
+     public function getCMSFields($collctnId, $recordType, $fieldCount) {
+       if (CMSRecords::isCMSRecord($recordType)) {
+
+        // get collection record so we can set the cms ID
+        $thisClctn = Collection::findorFail($collctnId);
+
+        $response = CMSRecords::findCMSHeader($recordType, $fieldCount);
+
+        // if only one item is found we return that result
+        if (count($response) == 1) {
```

Comment:
Is it really possible to get a CmsID that is null when you are specifically calling a Cms based dataset?  This seems a big illogical, if it is possible something else needs to be refactored to eliminate this factor.